alvr/server_openvr/src/tracking.rs

```diff
@@ -262,3 +262,44 @@ pub fn to_openvr_ffi_hand_skeleton(
 
     to_ffi_skeleton(&skeleton)
 }
+
+pub fn offset_motion(config: &HeadsetConfig, device_id: u64, motion: &mut DeviceMotion) {
+    let (t, r) = if let Switch::Enabled(controllers) = &config.controllers {
+        (
+            controllers.left_controller_position_offset,
+            controllers.left_controller_rotation_offset,
+        )
+    } else {
+        return;
+    };
```

Comment:
I'd also categorize this as unexpected behavior + passing in a too wide grained struct. This should just get the actual controller config directly, then it won't behave unexpectedly if the calling code changes. (Assuming controllers are disabled but we still want the data, e.g. to use them as body trackers, and thus want to apply the prediction.)

That then makes it so the calling code can move the filter before the map and just do an unwrap, or not do a filter but just do an if `some(config) ... else None` and use that as the decision. Either way I think that puts the responsibility more where it should be.